File: Data Structures & Algorithms/top-k-elements-in-list/submission-19.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        let bucket = Array.from({ length: nums.length + 1 }, () => []);
        let count = {};
        for (let n of nums) {
            count[n] = (count[n] ?? 0) + 1;
        }

        for (let n in count) {
            bucket[count[n]].push(n);
        }

        let res = [];
        for (let i = bucket.length - 1; i !== 0; i--) {
            for (let n of bucket[i]) {
                res.push(n);
                if (--k === 0) {
                    return res;
                }
            }
            
        }

        return [];
    }
}

File: Data Structures & Algorithms/is-anagram/submission-19.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        a = {}
        b = {}

        for c in s:
            if a.get(c):
                a[c] = a[c] + 1
            else:
                a[c] = 1;
        for c in t:
            if b.get(c):
                b[c] = b[c] + 1
            else:
                b[c] = 1;

        if len(a) != len(b):
            return False

        for c in b:
            if a.get(c) != b.get(c):
                return False

        return True
        
        